homeassistant/components/essent/sensor.py

```diff
@@ -0,0 +1,105 @@
+"""Support for Essent API."""
+from datetime import timedelta
+import xml.etree.ElementTree as ET
+
+from pyessent import PyEssent
+
+from homeassistant.const import ENERGY_KILO_WATT_HOUR, STATE_UNKNOWN
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+DOMAIN = 'essent'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(hours=1)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Essent platform."""
+    username = config['username']
+    password = config['password']
+    for meter in EssentBase(username, password).retrieve_meters():
+        for tariff in ['L', 'N', 'H']:
+            add_devices([EssentMeter(username, password, meter, tariff)])
+
+
+class EssentBase():
```

Comment:
I get it but I'd rather keep this separated so that each meter is an instance of a class that only checks stuff for that meter. Personal preference I suppose.

Reply:
Oh, I now realize that this authenticates and downloads data once for each meter and for each tariff, right?

We should probably optimize that to minimize the number of requests. That would be friendly towards the provider, especially once thousands of people install your sensor :-). This class could be used to cache the data.

Reply:
How does 5a2e8e3ee042b83d7035f739c1aeaf66d3a0f179 look?

Reply:
That looks good. Just give me a few minutes to digest.

@dgomes The throttle makes sure that we hit the external API just once per half hour, even when multiple sensors update.